lib/audit/types.go

```diff
@@ -17,18 +17,30 @@ limitations under the License.
 package audit
 
 import (
+	reg "sigs.k8s.io/k8s-container-image-promoter/lib/dockerregistry"
 	"sigs.k8s.io/k8s-container-image-promoter/lib/logclient"
 	"sigs.k8s.io/k8s-container-image-promoter/lib/remotemanifest"
 	"sigs.k8s.io/k8s-container-image-promoter/lib/report"
+	"sigs.k8s.io/k8s-container-image-promoter/lib/stream"
 )
 
+// GcrReadingFacility holds functions used to create streams for reading the
+// repository and manifest list.
+//
+// nolint[lll]
+type GcrReadingFacility struct {
```

Comment:
Nit: I don't love the name, but I'm not sure I understand it well enough to propose a better one.  `GcrReader`?